packages/nimble-components/src/rich-text/specs/README.md

```diff
@@ -107,6 +111,116 @@ Example usage of the `nimble-rich-text-editor` in the application layer is as fo
 </nimble-rich-text-editor>
 ```
 
+Additionally, this rich text editor also includes a feature for tagging or mentioning a user by inserting the **"@"** symbol within the editor.
+This action triggers a dropdown menu, allowing users to select a person from a list. To provide the rich text editor component with the
+necessary user information for populating this dropdown, users can pass user details through the configuration elements outlined below:
+
+```html
+<nimble-rich-text-editor>
+    <nimble-rich-text-mention-users>
+        <nimble-mapping-mention-user
+            key="user-id-1"
+            text="John Doe"
+        ></nimble-mapping-mention-user>
+        <nimble-mapping-mention-user
+            key="user-id-2"
+            text="Alice Smith"
+        ></nimble-mapping-mention-user>
+        <nimble-mapping-mention-user
+            key="user-id-3"
+            text="Bob Jones"
+        ></nimble-mapping-mention-user>
+    </nimble-rich-text-mention-users>
+</nimble-rich-text-editor>
+```
+
+The configuration element, `nimble-rich-text-mention-users`, consists of mapping elements that specify both the content to display
+in the dropdown list (i.e., the `text`) and the data to store in the markdown when extracting content from the editor (i.e., the `key`).
+These details are subsequently transformed into a map or an object, which is used to populate the options within the shadow root
+for the dropdown list of items.
+
+The `nimble-mapping-mention-user` extending from a base mapping class and utilize the same properties that base class offers.
+
+_Future Scope_:
+
+1. The `nimble-mapping-mention-user` can be used to get other user details like email, profile information etc., to display in the
+   dropdown list options.
+2. If there is a requirement to mention an issue (using **"#"**) or a pull requests (using **"!"**),
+   a new configuration component can be created and added as a child to the rich text editor. Below is an example of how
+   these elements would be used:
+
+```html
+<nimble-rich-text-editor>
+    <nimble-rich-text-mention-users>
+        <nimble-mapping-mention-user
+            key="user-id-1"
+            text="John Doe"
+        ></nimble-mapping-mention-user>
```

Comment:
Yes, it will have a `key` and `text` for every mention types but for other future properties (like email) for user mention, it will not be common for all mentions. It was one of the [comment](https://github.com/ni/nimble/pull/1594#discussion_r1360923257) above from Mert to have a distinct mapping component.